core/dbt/contracts/util.py

```diff
@@ -298,6 +298,8 @@ def upgrade_manifest_json(manifest: dict) -> dict:
         if "root_path" in doc_content:
             del doc_content["root_path"]
         doc_content["resource_type"] = "doc"
+    if "groups" not in manifest:
+        manifest["groups"] = {}
```

Comment:
Looks like this is a duplicate of lines 279-290: https://github.com/dbt-labs/dbt-core/pull/7007/files#diff-f009cc111693f698a64598745128407e594cb9679e14b4c40faf6c072699c897R279-R280

Reply:
Ooops. It was failing with "no groups field", but that was because I hadn't yet updated the version below which we need to update the manifest. Removed.